core/src/main/java/org/apache/druid/java/util/common/ByteBufferUtils.java

```diff
@@ -138,6 +144,45 @@ private static MethodHandle unmapJava9(MethodHandles.Lookup lookup) throws Refle
     return unmapper.bindTo(UnsafeUtils.theUnsafe());
   }
 
+  /**
```

Comment:
Could we also include a comment advising when `allocateDirect` should be used (especially in tests) rather than just using on-heap buffers with `ByteBuffer.allocate()`?

Reply:
Sure, I'll add the following. Let me know what you think.

```
  /**
   * Same as {@link ByteBuffer#allocateDirect(int)}, but returns a closeable {@link ResourceHolder} that
   * frees the buffer upon close.
   *
   * Direct (off-heap) buffers are an alternative to on-heap buffers that allow memory to be managed
   * outside the purview of the garbage collector. It's most useful when allocating big chunks of memory,
   * like processing buffers.
   */
```

Reply:
It looks good, thanks!